CMake/Fixups.cmake

```diff
@@ -0,0 +1,20 @@
+# TODO these should be in the submodules
+# Under Windows, we need Lua as DLL; on *nix we need it linked statically:
+if (WIN32)
+	target_compile_definitions(lua PUBLIC LUA_BUILD_AS_DLL)
+endif()
+
+# Let Lua use additional checks on its C API. This is only compiled into Debug builds:
+target_compile_definitions(lua PRIVATE LUA_USE_APICHECK)
+
+if(NOT MSVC AND "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm")
+	# mbed TLS uses the frame pointer's register in inline assembly for its bignum implementation:
+	# https://tls.mbed.org/kb/development/arm-thumb-error-r7-cannot-be-used-in-asm-here
+	target_compile_options(mbedcrypto PRIVATE -fomit-frame-pointer)
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+	# Generated file has old-style casts, missing prototypes, and deprecated declarations
+	set_source_files_properties("${CMAKE_SOURCE_DIR}/src/Bindings/Bindings.cpp" PROPERTIES COMPILE_OPTIONS -Wno-everything)
+	set_source_files_properties("${CMAKE_SOURCE_DIR}/src/IniFile.cpp" PROPERTIES COMPILE_OPTIONS -Wno-header-hygiene)
+endif()
```

Comment:
These aren't fixups for submodules.

Reply:
Indeed, more like fixups for questionable code

Reply:
Would you prefer they went in SetFlags?

Reply:
Source file properties should go in the folder's `CMakeLists.txt`.

Reply:
CMake can't propagate them... seriously :(

Reply:
Then put it at the top level `CMakeLists.txt` I guess.

Reply:
dun dun dun!